notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+    "\n",
+    "If you are running Druid on your laptop, the default configuration only provides 2 worker slots on the Middle Manager, so you can only run with `maxNumTasks=2` resulting in one controller and one worker. If you are using this notebook against a larger Druid cluster, feel free to experiment with higher values. Note that if `maxNumTasks` exceeds the available worker slots, the job will fail with a time out error because it waits for all the worker tasks to be active.\n",
+    "\n",
+    "##### rowsPerSegment\n",
+    "`rowsPerSegment` defaults to 3,000,000. You can adjust it to produce larger or smaller segments. \n",
+    "\n",
+    "This example shows how to set context parameters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55313e69-e63e-47d9-917c-2da87926ab73",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "REPLACE INTO \"wikipedia_events_4\" OVERWRITE ALL\n",
+    "WITH \"ext\" AS \n",
+    "(\n",
+    "    SELECT *\n",
+    "    FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\n",
+    "          \"uris\":[ \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\"\n",
+    "                 ]\n",
+    "         }',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    ") EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR))\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "request = druid.sql.sql_request( sql)         # init request object\n",
+    "request.add_context( 'rowsPerSegment', 20000) # setting it low to produce many segments\n",
+    "request.add_context( 'maxNumTasks', 2)        # can't go any higher in test env\n",
+    "\n",
+    "druid.display.run_task(request)\n",
+    "druid.sql.wait_until_ready('wikipedia_events_4')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ee87e942-000a-49d3-b915-4319519c4ed1",
+   "metadata": {},
+   "source": [
+    "With a `rowsPerSegment` of only 20,000, the same ingestion as before produces more segments. Open the [Druid console in the Data Sources view](http://localhost:8888/unified-console.html#datasources) to see the difference in segments between `wikipedia_events_3` and `wikipedia_events_4`.\n",
```

Comment:
Suggest run a SYS query to show this.

Reply:
not sure this helps.... I like showing it in the Druid console with a direct link to the appropriate view. Seems easier to understand....we should talk about this as a general notebook thingy...
did not change this.